### ISSUE TITLE
Issue with storing bool value

### ISSUE THREAD

**Issue by LocalHeroPro** [laravel/framework#37215]
- Laravel Version: 8.40.0
- PHP Version: 8.0.5
- Database Driver & Version:

### Description:
Try to store entry in database and have issue with bool value.

### Steps To Reproduce:
```php
Model::query()->create(['active' => true]);
```
get error:
```php
SQLSTATE[42804]: Datatype mismatch: 7 ERROR:  column "active" is of type boolean but expression is of type integer
LINE 1: ...", "version", "updated_at", "created_at") values ($1, $2, $3...
                                                             ^
HINT:  You will need to rewrite or cast the expression. (SQL: insert into "table_name" ("active", ...) values (1, ...) returning "id")
```

**Comment by hakanersu** [comment]
I'm currently getting this error;

- Laravel Version: 8.38.0
- PHP Version: 8.0.5
- Database Driver & Version:  (PostgreSQL) 13.2

Tried to cast it;

```php
protected $casts = [
        'is_active' => 'boolean',
];
```
It started after an update it can be related with PostgreSQL.

**Comment by driesvints** [comment]
This is a bug in PHP. Please see #37168

**Comment by LocalHeroPro** [comment]
How to deal with that in L8?

**Comment by driesvints** [comment]
You need to downgrade to PHP 8.0.4 or wait for 8.0.6. We can't do anything about this in Laravel itself.

**Comment by jstacho** [comment]
you can also workaround it in your application using a string representation value in your query:

`Model::query()->create(['active' => 't']);`
`Model::query()->create(['active' => 'true']);`
`Model::query()->create(['active' => '1']);`
`Model::query()->create(['active' => 'on']);`
`Model::query()->create(['active' => 'yes']);`

see https://www.postgresql.org/docs/current/datatype-boolean.html

**Comment by LocalHeroPro** [comment]
`8.0.5-2+ubuntu20.04.1+deb.sury.org+1` resolving the issue.

**Comment by njoguamos** [comment]
I was able to fix this by downgrading PHP 8.0.5 to PHP 8.0.3_1 via brew.

For those interested, [follow this instruction](https://gist.github.com/sawant/6ea10e45de7d09e966a1dc6afea3bffb)

I will upgrade to PHP 8.0.6 once it is released.
